feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c

```diff
@@ -302,12 +303,17 @@ static void vif_config_security_set(struct blob_buf *b,
 		blobmsg_add_bool(b, "ieee80211w", 1);
 		if (vif_crypto[i].enterprise) {
 			acct_interval = 0;
-			blobmsg_add_string(b, "auth_server",
-					   SCHEMA_KEY_VAL(vconf->security, SCHEMA_CONSTS_SECURITY_RADIUS_IP));
-			blobmsg_add_string(b, "auth_port",
-					   SCHEMA_KEY_VAL(vconf->security, SCHEMA_CONSTS_SECURITY_RADIUS_PORT));
-			blobmsg_add_string(b, "auth_secret",
-					   SCHEMA_KEY_VAL(vconf->security, SCHEMA_CONSTS_SECURITY_RADIUS_SECRET));
+			auth_server = SCHEMA_KEY_VAL(vconf->security, SCHEMA_CONSTS_SECURITY_RADIUS_IP);
+			auth_port   = SCHEMA_KEY_VAL(vconf->security, SCHEMA_CONSTS_SECURITY_RADIUS_PORT);
+			auth_secret = SCHEMA_KEY_VAL(vconf->security, SCHEMA_CONSTS_SECURITY_RADIUS_SECRET);
+			LOGT("%s: Server IP %s port %s secret %s", vconf->if_name, auth_server, auth_port, auth_secret);
+			if (!auth_server[0] || !auth_port[0] || !auth_secret[0]) {
```

Comment:
Aren't all of them mandatory? If yes, &&?

Reply:
The accounting entries are not mandatory.

Reply:
The first byte of each string will be "/0" if it is empty.  This evaluate to zero so the ! causes 1 and the error will be triggered if any string is empty.